helm/charts/determined/templates/master-config.yaml

```diff
@@ -57,7 +57,7 @@ stringData:
 
     security:
       {{- if .Values.initialUserPassword }}
-      initial_user_password: {{ .Values.initialUserPassword | quote }}
+      initial_user_password: {{ coalesce .Values.initialUserPassword .Values.defaultPassword | quote }}
       {{- end }}
```

Comment:
please check if anything has changed here:
https://docs.determined.ai/latest/reference/deploy/master-config-reference.html#initial-user-password

Reply:
I don't think anything should change there -- that's from #8851 which describes the structure that this configuration file is using. In Helm's case, there was previous support for a `.Values.defaultPassword` that would launch a sidecar. Before that, the flow for configuring the admin password was something like:

1. Launch a cluster with empty admin password
2. Wait a bit for the helm `postInstall` hook to fire
3. If `defaultPassword` is set,
  a. authorize as admin using empty password
  b. change the admin and determined passwords to `.defaultPassword`

After #8851 and #9074 , the flow is:

1. Launch a cluster; if `.Values.initialUserPassword` is set, use that for admin/determined users, otherwise allow empty passwords
2. Wait a bit for the helm `postInstall` hook to fire
3. If `defaultPassword` is set,
  a. authorize as admin using no password (**fails if `.Values.initialUserPassword` was also set, since admin has a non-empty password!**)
  b. try to change the admin and determined passwords to `.defaultPassword`

By changing this line and deleting the sidecar, the intention (which I've manually tested, but probably not exhaustively!) is that the new flow is:

1. Launch a cluster; if `.Values.initialUserPassword` is set, use that for admin/determined users; otherwise use `.Values.defaultPassword`; if both are blank, an attempt is made with a blank password (which will eventually fail because of all the other changes we're making around this).

I hope this is helpful/informative and not just over-explaining things you already knew! If there's something about this that needs to be communicated and isn't already handled in your #9101 I could use some help writing copy in the style most consistent with Determined docs. 😄